core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/conditional/DecodeExpression.kt

```diff
@@ -0,0 +1,42 @@
+package ch.ergon.dope.resolvable.expression.unaliased.type.conditional
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.expression.UnaliasedExpression
+import ch.ergon.dope.resolvable.operator.FunctionOperator
+import ch.ergon.dope.validtype.ValidType
+
+class DecodeExpression<T : ValidType, U : ValidType>(
+    private val expression: UnaliasedExpression<T>,
+    private val searchResult: SearchResult<T, U>,
+    private vararg val searchResults: SearchResult<T, U>,
+    private val default: UnaliasedExpression<U>? = null,
+) : TypeExpression<U>, FunctionOperator {
+    override fun toDopeQuery(): DopeQuery {
+        val expressionDopeQuery = expression.toDopeQuery()
+        val searchResultDopeQuery = searchResult.toDopeQuery()
+        val searchResultsDopeQuery = searchResults.map { it.toDopeQuery() }.toTypedArray()
+        val defaultDopeQuery = default?.toDopeQuery()
+        return DopeQuery(
+            queryString = toFunctionQueryString(
+                "DECODE",
+                expressionDopeQuery,
+                searchResultDopeQuery,
+                *searchResultsDopeQuery,
+                defaultDopeQuery,
+            ),
+            parameters = expressionDopeQuery.parameters +
+                searchResultsDopeQuery.fold(searchResultDopeQuery.parameters) {
+                        expressionParameters, expression ->
+                    expressionParameters + expression.parameters
+                } + defaultDopeQuery?.parameters.orEmpty(),
+        )
+    }
+}
+
+fun <T : ValidType, U : ValidType> decode(
```

Comment:
can you add a buildTest as well? And how does the user create a SearchResult, shouldn't we add an extension-function as well (like for the set-assignment)>?

Reply:
I can definitely add a few build tests. I'm a bit confused with your second question, I added extension functions in the `SearchResult.kt` file, do you mean those or specifically different ones?

Reply:
Ah I misssed that file somehow, ignore my second question then